specification/botservice/resource-manager/Microsoft.BotService/preview/2017-12-01/botservice.json

```diff
@@ -1618,6 +1982,202 @@
         "siteName"
       ]
     },
+    "ConnectionItemName": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Connection Item name that has been added in the API",
+          "readOnly": true
+        }
+      },
+      "description": "The display name of a connection Item Setting registered with the Bot"
+    },
+    "ConnectionSettingParameter": {
+      "properties": {
+        "key": {
+          "type": "string",
+          "description": "Key for the Connection Setting Parameter.",
+          "readOnly": true
+        },
+        "value": {
+          "type": "string",
+          "description": "Value associated with the Connection Setting Parameter.",
+          "readOnly": true
+        }
+      },
+      "description": "Extra Parameter in a Connection Setting Properties to indicate service provider specific properties"
+    },
+    "ConnectionSettingProperties": {
+      "properties": {
+        "clientId": {
+          "type": "string",
+          "description": "Client Id associated with the Connection Setting."
+        },
+        "clientSecret": {
+          "type": "string",
+          "description": "Client Secret associated with the Connection Setting"
+        },
+        "scopes": {
+          "type": "string",
+          "description": "Scopes associated with the Connection Setting"
+        },
+        "id": {
+          "type": "string",
+          "description": "Id associated with the Connection Setting"
+        },
+        "name": {
+          "type": "string",
+          "description": "Name associated with the Connection Setting"
+        },
+        "serviceProviderId": {
+          "type": "string",
+          "description": "Service Provider Id associated with the Connection Setting"
+        },
+        "serviceProviderDisplayName": {

```

Comment:
Why do you need both serviceProviderId and serviceProviderDisplayname to be provided? serviceProviderId should be sufficient.

Reply:
The serviceproviderdisplayname is needed by the botservice azure portal blade to display the service providers available as a dropdown, when they add a connection. it makes better ui experience to have the display name than to have just the id.